pytorch_lightning/plugins/checkpoint/torch.py

```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.checkpoint.checkpoint import CheckpointIOPlugin
+from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.utilities.cloud_io import atomic_save
+from pytorch_lightning.utilities.cloud_io import load as pl_load
+
+
+class TorchCheckpointIOPlugin(CheckpointIOPlugin):
+    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
+        # dump states as a checkpoint dictionary object
+        try:
+            # write the checkpoint dictionary on the file
+            atomic_save(checkpoint, filepath)
+        except AttributeError as err:
+            key = pl.LightningModule.CHECKPOINT_HYPER_PARAMS_KEY
+            checkpoint.pop(key, None)
+            rank_zero_warn(f"Warning, `{key}` dropped from checkpoint. An attribute is not picklable: {err}")
+            atomic_save(checkpoint, filepath)
+
+    def load_checkpoint_file(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
+        return pl_load(checkpoint_path, map_location=(lambda storage, loc: storage))
```

Comment:
should we directly use fsspec here? https://github.com/PyTorchLightning/pytorch-lightning/blob/e54180363663b690367f0051484650fcd12e5de2/pytorch_lightning/utilities/cloud_io.py#L36-L38

the handling here is very hacky, which the checkpoint plugin should have much more control over this

Reply:
The checkpoint plugin can be overridden to control this right?

Which part of the cloud io code is hacky? I'm not familiar with this part of the codebase so might need some assistance here!